pulsar/consumer_regex.go

```diff
@@ -343,7 +343,7 @@ func (c *regexConsumer) unsubscribe(topics []string) {
 }
 
 func (c *regexConsumer) topics() ([]string, error) {
-	topics, err := c.client.namespaceTopics(c.namespace)
+	topics, err := c.client.lookupService.GetTopicsOfNamespace(c.namespace, internal.Persistent)
```

Comment:
For regex are only persistent topics supported?

Reply:
not actually for other client like Java, but this PR does not tend to add `non-persistence` and `all` supporting, will do it in another PR.